libsabledb/src/commands/generic_commands.rs

```diff
@@ -170,6 +174,105 @@ impl GenericCommands {
         };
         Ok(Some(md.value_type()))
     }
+
+    async fn expire(
+        client_state: Rc<ClientState>,
+        command: Rc<RedisCommand>,
+        response_buffer: &mut BytesMut,
+    ) -> Result<(), SableError> {
+        // at least 3 arguments
+        check_args_count!(command, 3, response_buffer);
+
+        let db_id = client_state.database_id();
+        let builder = RespBuilderV2::default();
+        let generic_db = GenericDb::with_storage(&client_state.store, db_id);
+
+        // EXPIRE key seconds [NX | XX | GT | LT]
+        let mut iter = command.args_vec().iter();
+        // skip "expire"
+        iter.next();
+
```

Comment:
I almost missed it..., but you need to extract the key first, lock it and only then you can access the database.
Since you are doing here "get" followed by "put", you need to obtain an "exclusive" lock on the key - so the keys is persistent between the calls

```rust
let _unused = LockManager::lock_user_key_exclusive(&key, client_state.database_id());
```